src/endpoint/s3/s3_rest.js

```diff
@@ -446,18 +412,18 @@ function parse_op_name(req) {
 function handle_error(req, res, err) {
     var s3err =
         ((err instanceof S3Error) && err) ||
-        new S3Error(RPC_ERRORS_TO_S3[err.rpc_code] || S3Error.InternalError);
+        new S3Error(S3Error.RPC_ERRORS_TO_S3[err.rpc_code] || S3Error.InternalError);
 
     if (s3err.code === 'MalformedPolicy') {
         s3err.message = err.message;
         s3err.detail = err.rpc_data.detail;
     }
 
     if (s3err.rpc_data) {
-        if (s3err.rpc_data.etag) {
+        if (s3err.rpc_data.etag && !res.headersSent) {
```

Comment:
I would avoid silent ignore at this stage, at least until we're sure that's OK:
```suggestion
        if (s3err.rpc_data.etag) {
            if (res.headersSent) {
                dbg.log0('...');
            } else {
```

Same for the last_modified header below too.